### PR TITLE
Graduate @lit-labs/task out to @lit/task

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: packages/labs/task/src/task.ts [hunk 1]
```diff
@@ -3,441 +3,16 @@
...
-  : undefined;
-
-export const shallowArrayEquals = <T extends ReadonlyArray<unknown>>(
-  oldArgs: T,
-  newArgs: T
-) =>
-  oldArgs === newArgs ||
-  (oldArgs.length === newArgs.length &&
-    oldArgs.every((v, i) => !notEqual(v, newArgs[i])));
+export {
```
Comment (augustjk) [inline packages/labs/task/src/task.ts:7]: should we re-export with `@deprecated`? or we leave as is and just mark the package as deprecated in the npm registry?

Comment (rictic) [reply packages/labs/task/src/task.ts:11]: Yeah, good idea!

File: packages/task/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
+# Change Log
+
+## 3.0.1
```
[the remark this answers is not in the document]
Comment (rictic) [reply packages/task/CHANGELOG.md]: Great point!

File: rollup-common.js [hunk 3]
```diff
@@ -23,17 +23,18 @@ const PACKAGE_CLASS_PREFIXES = {
...
-  '@lit-labs/react': '_$G',
-  '@lit-labs/scoped-registry-mixin': '_$H',
-  '@lit-labs/ssr-client': '_$I',
-  '@lit-labs/task': '_$J',
-  '@lit-labs/router': '_$K',
-  '@lit-labs/observers': '_$L',
-  '@lit-labs/context': '_$M',
-  '@lit-labs/vue-utils': '_$N',
-  '@lit-labs/preact-signals': '_$O',
+  '@lit/task': '_$E',
```
[the remark this answers is not in the document]
Comment (rictic) [reply rollup-common.js]: Hm, I didn't think of that. Accessing private fields from other versions is fraught, but it should reduce surprising behavior at least if we keep them consistent. Ideally I guess we'd have unique prefixes for each release, but that would bloat binary sizes a bit, and probably not worth the additional engineering cost..

**Review by augustjk — APPROVED** [review]
🎓 🎊

File: packages/task/package.json [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      "output": []
+    },
+    "test": {
+      "dependencies": [
+        "test:dev",
+        "test:prod"
+      ]
+    },
+    "test:dev": {
+      "command": "MODE=dev node ../tests/run-web-tests.js \"development/**/*_test.js\" --config ../tests/web-test-runner.config.js",
```
Comment (AndrewJakubowicz) [inline packages/task/package.json:125]: Nit: `MODE=dev` and `MODE=prod` (in the next test) could be moved into wireit `env`, but non blocking.

**Review by AndrewJakubowicz — APPROVED** [review]
Awesome! LGTM!

File: lit-next.code-workspace [hunk 5]
```diff
@@ -20,6 +20,10 @@
       "name": "ssr",
       "path": "packages/labs/ssr"
     },
+    {
```
Comment (justinfagnani) [inline lit-next.code-workspace:23]: VS Code has gotten a lot better about working with nested packages that aren't workspace roots, so I don't know if we need to add every package here anymore. I'm also fine with it.

File: packages/labs/task/README.md [hunk 6]
```diff
@@ -2,85 +2,8 @@
 
 A controller for Lit that renders asynchronous tasks.
 
-## Overview
+This package has graduated from labs! It is now available as `@lit/task`. This package will no longer be updated, but it will continue to work and get updates through the ^1.0.0 version range of `@lit/task` because it's now just a re-export of that package. This should reduce duplication of code while everyone migrates their imports and dependencies.
```
Comment (justinfagnani) [inline packages/labs/task/README.md:5]: I think the "it's now just a re-export of that package" part should com earlier, then mention this package is a proxy and won't be updated.

We might want a headline here too.

File: packages/labs/task/package.json [hunk 7]
```diff
@@ -57,12 +56,14 @@
         "build:ts:types",
         "build:rollup",
         "../../lit:build",
-        "../../reactive-element:build"
+        "../../reactive-element:build",
```
Comment (justinfagnani) [inline packages/labs/task/package.json:59]: do we need lit and reactive-element anymore?

File: packages/labs/task/package.json [hunk 8]
```diff
@@ -156,8 +146,9 @@
     "@lit-internal/scripts": "^1.0.1-pre.0",
     "lit": "^3.0.0-pre.0"
   },
+  "#versionComment": "We depend on <2.0.0 of task because we want to be compatible with 0.x.x versions, as well as the 1.x.x versions, as they're intended to all be compatible with the 3.x.x API of lit-labs/task before it graduated out of labs.",
   "dependencies": {
-    "@lit/reactive-element": "^2.0.0-pre.0"
+    "@lit/task": "<2.0.0"
```
Comment (justinfagnani) [inline packages/labs/task/package.json:151]: The only issue with doing it this way is that it's slightly harder to pin to a specific version. In that case though the user can use `@lit/task` directly, or use npm's new-ish resolution overrides.

File: packages/labs/task/src/deep-equals.ts [hunk 9]
```diff
@@ -4,121 +4,19 @@
...
- *    - have the same constructor
- *    - have same set of own property names
- *    - have each own property be deeply equal
- *  - Arrays, Maps, Sets, and RegExps
- *  - Objects with custom valueOf() (ex: Date)
- *  - Objects with custom toString() (ex: URL)
- *
- * Important: Objects must be free of cycles, otherwise this function will
- * run infinitely!
+ * You can import directly from @lit/task/deep-equals.js now.
```
Comment (justinfagnani) [inline packages/labs/task/src/deep-equals.ts:8]: Does this deprecate imports of this file by being on the export? Or does that need to be in a `@fileoverview` or `@module` comment?

File: packages/task/package.json [hunk 10]
```diff
@@ -0,0 +1,165 @@
+{
+  "name": "@lit/task",
+  "version": "0.1.0",
```
Comment (justinfagnani) [inline packages/task/package.json:3]: This could be `0.0.0` to make it clear it hasn't been released yet.

File: packages/task/rollup.config.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {litProdConfig} from '../../rollup-common.js';
+import {createRequire} from 'module';
+
+export default litProdConfig({
+  packageName: createRequire(import.meta.url)('./package.json').name,
+  entryPoints: ['deep-equals', 'index', 'task'],
+  external: ['@lit/reactive-element'],
```
Comment (justinfagnani) [inline packages/task/rollup.config.js:13]: In the future I think we should change all of these to `external: (id) => !(id.startsWith('.') || id.startsWith('/')),`

**Review by justinfagnani — APPROVED** [review]
🎉

File: packages/labs/task/package.json (hunk 7 above)
Comment (rictic) [reply packages/labs/task/package.json:59]: For tests, yes

File: packages/labs/task/package.json (hunk 8 above)
Comment (rictic) [reply packages/labs/task/package.json:151]: Package locks are the solution to this I think

File: packages/labs/task/src/deep-equals.ts (hunk 9 above)
Comment (rictic) [reply packages/labs/task/src/deep-equals.ts:8]: I tried those techniques as well as a couple others, and none worked to deprecate the whole module

File: packages/task/package.json (hunk 10 above)
Comment (rictic) [reply packages/task/package.json:3]: +1

File: packages/task/rollup.config.js (hunk 11 above)
Comment (rictic) [reply packages/task/rollup.config.js:13]: +1

File: lit-next.code-workspace (hunk 5 above)
Comment (rictic) [reply lit-next.code-workspace:23]: +1, but not in this PR

File: packages/labs/task/README.md (hunk 6 above)
Comment (rictic) [reply packages/labs/task/README.md:5]: +1, done

File: packages/tests/package.json [hunk 12]
```diff
@@ -33,10 +34,39 @@
...
+      "files": [],
+      "output": [],
+      "env": {
+        "BROWSERS": {
+          "external": true
+        }
+      }
+    },
+    "test:common:prod": {
+      "command": "MODE=prod node ./run-web-tests.js",
```
[the remark this answers is not in the document]
Comment (rictic) [reply packages/tests/package.json:56]: Good point, done

File: packages/task/package.json (hunk 4 above)
Comment (rictic) [reply packages/task/package.json:125]: Done

**Pull request merged**